src/test/java/ch/saunah/saunahbackend/service/SaunaServiceTest.java

```diff
@@ -153,4 +162,75 @@ private void checkSaunaFields(SaunaTypeBody saunaTypeBody, Sauna sauna) {
         assertEquals(saunaTypeBody.getType(), sauna.getType());
         assertEquals(saunaTypeBody.getPlz(), sauna.getPlz());
     }
+
+
+    /**
+     * This test checks if images can be added to a sauna successfully.
+     */
+    @Test
+    @DirtiesContext(methodMode = DirtiesContext.MethodMode.AFTER_METHOD)
+    void addSaunaImage() {
+        assertThrows(NotFoundException.class, ()-> saunaService.addSaunaImages(0, null));
+        saunaService.addSauna(saunaTypeBody);
+        Sauna sauna = saunaService.getAllSauna().get(0);
+        assertThrows(NullPointerException.class, ()-> saunaService.addSaunaImages(sauna.getId(), null));
+        List<SaunaImage> images = saunaService.getSaunaImages(sauna.getId());
+        assertTrue(images.isEmpty());
+        saunaService.addSaunaImages(sauna.getId(), new ArrayList<>());
+        images = saunaService.getSaunaImages(sauna.getId());
+        assertTrue(images.isEmpty());
+        List<MultipartFile> testfiles = new ArrayList<>();
+        testfiles.add(new MockMultipartFile("test", new byte[0]));
+        saunaService.addSaunaImages(sauna.getId(), testfiles);
+        assertEquals(testfiles.size(), saunaService.getSaunaImages(sauna.getId()).size());
+        testfiles.clear();
+        testfiles.add(new MockMultipartFile("test", new byte[0]));
+        testfiles.add(new MockMultipartFile("test", new byte[0]));
+        saunaService.addSaunaImages(sauna.getId(), testfiles);
+        int savedImagesCount = saunaService.getSaunaImages(sauna.getId()).size();
+        assertEquals(3, savedImagesCount);
+    }
+
+    /**
+     * This test checks if an added image can be removed from the database.
+     */
+    @Test
+    @DirtiesContext(methodMode = DirtiesContext.MethodMode.AFTER_METHOD)
+    void removeSaunaImage() {
+        assertThrows(NotFoundException.class, () -> saunaService.removeSaunaImage(1));
+        saunaService.addSauna(saunaTypeBody);
+        Sauna sauna = saunaService.getAllSauna().get(0);
+        List<MultipartFile> testfiles = new ArrayList<>();
+        testfiles.add(new MockMultipartFile("test", new byte[0]));
+        testfiles.add(new MockMultipartFile("test", new byte[0]));
+        saunaService.addSaunaImages(sauna.getId(), testfiles);
+        assertEquals(testfiles.size(), saunaService.getSaunaImages(sauna.getId()).size());
+        List<SaunaImage> images = saunaService.getSaunaImages(sauna.getId());
+        SaunaImage saunaImage = images.get(0);
+        saunaService.removeSaunaImage(saunaImage.getId());
+        assertEquals(1, saunaService.getSaunaImages(sauna.getId()).size());
+    }
+
+    /**
+     * This test checks if all saunas can be found that exist in the database
+     */
+    @Test
+    @DirtiesContext(methodMode = DirtiesContext.MethodMode.AFTER_METHOD)
+    void getSaunaImage() throws IOException {
```

Comment:
Javadoc you probably meant "Sauna Images" instead of just sauna (sorry for nitpicking)